src/evm/transaction-count.ts

```diff
@@ -0,0 +1,38 @@
+import * as node from '@api3/airnode-node';
+import { ethers } from 'ethers';
+import isNil from 'lodash/isNil';
+import { SponsorWalletTransactionCount } from '../types';
+import { retryGo } from '../utils';
+import { deriveSponsorWallet } from '../wallet';
+
+export const getSponsorWalletAndTransactionCount = async (
+  airnodeWallet: ethers.Wallet,
+  provider: ethers.providers.Provider,
+  currentBlock: number,
+  sponsor: string
+): Promise<node.LogsData<SponsorWalletTransactionCount | null>> => {
+  // Derive sponsorWallet address
+  // TODO: switch to node.evm.deriveSponsorWallet when @api3/airnode-node allows setting the `protocolId`
+  const sponsorWallet = deriveSponsorWallet(
+    airnodeWallet.mnemonic.phrase,
+    sponsor,
+    '2' // TODO: should this be in a centralized enum somewhere (api3/airnode-protocol maybe)?
+  ).connect(provider);
+
+  // Fetch sponsorWallet transaction count
+  const [errorGetTransactionCount, transactionCount] = await retryGo(() =>
+    provider.getTransactionCount(sponsorWallet.address, currentBlock)
+  );
+  if (errorGetTransactionCount || isNil(transactionCount)) {
+    const message = 'Failed to fetch the sponsor wallet transaction count';
+    const log = node.logger.pend('ERROR', message, errorGetTransactionCount);
+    return [[log], null];
+  }
+
+  const message = `Sponsor wallet ${sponsorWallet.address.replace(
```

Comment:
Maybe these interpolated function calls should be done separately? This is kinda difficult to read